# TUGAS 3 : PRAKTIKUM ALGORITMA PEMROGRAMAN

NAMA : MOHAMMAD IQBAL SURYA RAMADHAN

NIM  : 210411100002

MATA KULIAH : ALGORITMA DAN PEMROGRAMAN DASAR

In [ ]:
# nomor 1

import math
radius = 6371.01
x1, y1 = eval(input("Masukkan titik 1 (latitude, longitude) dalam derajat: "))
x2, y2 = eval(input("Masukkan titik 2 (latitude, longitude) dalam derajat: "))
d= radius * math.acos(math.sin(math.radians(x1))*math.sin(math.radians(x2))+math.cos(math.radians(x1))*math.cos(math.radians(x2))*math.cos(math.radians(y1-y2)))
print("Jarak antara dua titik ini adalah", d,"km" )

Jarak antara dua titik ini adalah 10691.79183231593 km


nomor 1

(Geometri: jarak lingkaran besar)
Jarak lingkaran besar adalah jarak antara dua titik pada permukaan bumi. Katakanlah
(x1, y1) dan (x2, y2) menjadi garis lintang dan garis bujur geografis dari dua titik.
Jarak lingkaran besar antara dua titik dapat dihitung dengan menggunakan rumus
berikut:

Buatlah program yang meminta pengguna untuk memasukkan garis lintang dan garis
bujur dari dua titik di bumi dalam derajat dan tampilkan jarak lingkaran besarnya.
Jari-jari bumi rata-rata adalah 6.371,01 km. Perhatikan bahwa Anda perlu mengubah
derajat menjadi radian menggunakan fungsi math.radian karena fungsi trigonometri
Python menggunakan radian. Derajat lintang dan bujur dalam rumus adalah untuk
utara dan barat. Gunakan negatif untuk menunjukkan derajat selatan dan timur.
Berikut adalah contoh saat program berjalan:
Masukkan titik 1 (latitude, longitude) dalam derajat: 39.55, -116.25
Masukkan titik 2 (latitude, longitude) dalam derajat: 41.5, 87.37 Jarak antara dua titik
ini adalah 10691.79183231593 km

nomor 2

(Geografi: Memperkirakan luas)
Carilah lokasi GPS dari kota Atlanta, Georgia; Orlando, Florida; Savannah, Georgia;
dan Charlotte, North Carolina dari www.gps-data-team.com/map/ dan hitung
perkiraan luas yang dilingkupi oleh keempat kota ini.
Catatan:
• Gunakan rumus pada masalah 5 di atas untuk menghitung jarak antara 2 kota
• Bagi polygon menjadi dua segitiga dan gunakan rumus berikut untuk
menghitung luas segitiga tersebut

In [ ]:
# nomor 2
#codee

import math

Atlanta_x, Atlanta_y = 33.753746, -84.386330
Savannah_x, Savannah_y = 28.5383355, -81.3792364999
Orlando_x, Orlando_y = 32.0835407, -81.09983419999998
Charlotte_x, Charlotte_y = 35.2270869, -80.84312669999997


avg = 6371.01

#jarak antara Atlanta dan Savannah di permukaan bumi
AS = avg * math.acos (
    math.sin(math.radians(Atlanta_x))
    * math.sin(math.radians(Savannah_x))
    + math.cos(math.radians(Atlanta_x))
    * math.cos(math.radians(Savannah_x))
    * math.cos(math.radians(Atlanta_y - Savannah_y)))

#jarak antara Atlanta dan Charlotte di permukaan bumi
AC = avg * math.acos (
    math.sin(math.radians(Atlanta_x))
    * math.sin(math.radians(Charlotte_x))
    + math.cos(math.radians(Atlanta_x))
    * math.cos(math.radians(Charlotte_x))
    * math.cos(math.radians(Atlanta_y - Charlotte_y)))

#jarak antara Savannah dan Charlotte di bumi
SC = avg * math.acos (
    math.sin(math.radians(Savannah_x))
    * math.sin(math.radians(Charlotte_x))
    + math.cos(math.radians(Savannah_x))
    * math.cos(math.radians(Charlotte_x))
    * math.cos(math.radians(Savannah_y - Charlotte_y)))

#jarak antara Atlanta dan Orlando di permukaan bumi
AO = avg* math.acos (
    math.sin(math.radians(Atlanta_x))
    * math.sin(math.radians(Orlando_x))
    + math.cos(math.radians(Atlanta_x))
    * math.cos(math.radians(Orlando_x))
    * math.cos(math.radians(Atlanta_y - Orlando_y)))

#jarak antara Savannah dan Orlando di permukaan bumi
SO = avg * math.acos (
      math.sin(math.radians(Savannah_x))
    * math.sin(math.radians(Orlando_x))
    + math.cos(math.radians(Savannah_x))
    * math.cos(math.radians(Orlando_x))
    * math.cos (math.radians(Savannah_y - Orlando_y)))
#luas segitiga antara Atlanta, Savannah dan Charlotte
s = (AS + AC + SC) / 2 
area1 = (s * (s - AS) * (s - AC) * (s - SC)) ** 0.5
print('luas segitiga 1 :', area1)

#luas segitiga antara Atlanta, Savannah dan Orlando
s = (AS + AO + SO) / 2
area2 = ( s * (s - AS) * (s - AO) * (s - SO)) ** 0.5
print('luas segitia 2 :', area2)

area = area1 + area2

#display area poligon yang dibuat oleh 4 kota
print("Perkiraan luas daerah yang dilingkupi oleh 4 kota adalah : " + str(area), "sq.km")


luas segitiga 1 : 117570.91834622984
luas segitia 2 : 62499.43328415301
Perkiraan luas daerah yang dilingkupi oleh 4 kota adalah : 180070.35163038285 sq.km


(Geometri: luas polygon beraturan)
Poligon beraturan adalah poligon bersisi n di mana semua sisinya sama panjang dan
semua sudutnya memiliki derajat yang sama (yaitu, poligonnya sama sisi dan sama
sudut). Rumus untuk menghitung luas poligon beraturan adalah

Di sini, s adalah panjang sisi.

Tulis program yang meminta pengguna untuk memasukkan jumlah sisi dan panjang sisi
tersebut dari suatu poligon beraturan dan tampilkan luasnya.
Berikut ini adalah contoh eksekusinya:
Masukkan jumlah sisi: 5
Masukkan panjang sisi: 6.5
Luas poligon adalah 73.69017017488385

In [ ]:
# nomor 3

import math
n = eval(input("Masukkan jumlah sisi: "))
s = eval(input("Masukkan panjang sisi: "))
area = (n*(s**2))/(4*math.tan(math.pi/n))
print("Luas poligon adalah ",area)

Luas poligon adalah  72.69017017488386


(Aplikasi keuangan: unit moneter)

Tulis ulang (revisi) kode program ComputeChange.py di atas, untuk memperbaiki
kemungkinan hilangnya akurasi saat mengonversi nilai float ke nilai int. Masukkan input
sebagai bilangan bulat yang dua digit terakhirnya mewakili sen. Misalnya, input 1156
mewakili
11 dolar dan 56 sen.

In [ ]:
# nomor 4

# Receive the amount
amount = int(input("Masukkan nominal uang: "))
remainingAmount = amount

# Find the number of cents in the remaining amount
numberofCents = remainingAmount % 100

# Find the number of one dollars
numberOfOneDollars = remainingAmount // 100
remainingAmount = remainingAmount % 100

# Find the number of quarters in the remaining amount
numberOfQuarters = remainingAmount // 25
remainingAmount = remainingAmount % 25

# Find the number of dimes in the remaining amount
numberOfDimes = remainingAmount // 10
remainingAmount = remainingAmount % 10

# Find the number of nickels in the remaining amount
numberOfNickels = remainingAmount // 5
remainingAmount = remainingAmount % 5

# Find the number of pennies in the remaining amount
numberOfPennies = remainingAmount

# Display the results
print("Nominal sebesar", amount, "terdiri dari\n",
"\t", numberOfOneDollars, "dollars\n",
"\t", numberOfQuarters, "quarters\n",
"\t", numberOfDimes, "dimes\n",
"\t", numberOfNickels, "nickels\n",
"\t", numberOfPennies, "pennies")
print(numberOfOneDollars, "dolar", numberofCents, "sen")

Nominal sebesar 1156 terdiri dari
 	 11 dollars
 	 2 quarters
 	 0 dimes
 	 1 nickels
 	 1 pennies
11 dolar 56 sen


In [ ]:
# nomor 4

# Terima jumlah
jumlah = int(input("Masukkan jumlah nominal uang: "))
jumlahTersisa = jumlah

# Temukan jumlah sen dalam jumlah yang tersisa
jumlahSen = jumlahTersisa % 100

# Temukan jumlah satu dolar
jumlahSatuDollars = jumlahTersisa // 100
jumlahTersisa = jumlahTersisa % 100

# Temukan jumlah perempat dalam jumlah yang tersisa
jumlahPerempat = jumlahTersisa // 25
jumlahTersisa = jumlahTersisa % 25

# Temukan jumlah uang receh dalam jumlah yang tersisa
jumlahUangReceh = jumlahTersisa // 10
jumlahTersisa = jumlahTersisa % 10

# Temukan jumlah sen dalam jumlah yang tersisa
jumlahNikel = jumlahTersisa // 5
jumlahTersisa = jumlahTersisa % 5

# Temukan jumlah sen dalam jumlah yang tersisa
jumlahPenny = jumlahTersisa

# Tampilkan hasilnya
print("Nominal sebesar", jumlah, "terdiri dari\n",
"\t", jumlahSatuDollars, "dolar\n",
"\t", jumlahPerempat, "perempat\n",
"\t", jumlahUangReceh, "uang receh\n",
"\t", jumlahNikel, "nickel\n",
"\t", jumlahPenny, "Penny")
print(jumlahSatuDollars, "dolar", jumlahSen, "sen")

Nominal sebesar 1156 terdiri dari
 	 11 dolar
 	 2 perempat
 	 0 uang receh
 	 1 nickel
 	 1 Penny
11 dolar 56 sen
